scripts/cvmfsexec-osg-wrapper

```diff
@@ -21,27 +21,29 @@
 #
 # Then set CVMFSEXEC_REPOS in the environment to a comma-or-space-separated
 # list of CVMFS repos to mount.
+#
+# Other variables:
+# - CVMFS_HTTP_PROXY: the proxy to use for CVMFS
+# - UMOUNTREPO_UNMOUNT_ALL:
+#       no effect unless using `mountrepo`; set this to unmount all mounted
+#       repos before and after.  This makes cleanup more reliable but may
+#       conflict if multiple copies of this script are running, so only use
+#       it with whole-machine jobs.
 
 CVMFSEXEC_REPOS=$(tr -s ',' ' ' <<<"${CVMFSEXEC_REPOS-}")
 CVMFS_HTTP_PROXY=${CVMFS_HTTP_PROXY-}
-
-if [[ -z $CVMFSEXEC_REPOS ]]; then
-    exec "$@"
-fi
-
-if [[ -z $CVMFS_HTTP_PROXY ]]; then
-    if [[ ${TACC_SYSTEM-} = stampede2 ]]; then
-        CVMFS_HTTP_PROXY=http://login5.stampede2.tacc.utexas.edu:3128/
-    fi
-fi
-
+UMOUNTREPO_UNMOUNT_ALL=${UMOUNTREPO_UNMOUNT_ALL-}
```

Comment:
```suggestion
UMOUNTREPO_UNMOUNT_ALL=${UMOUNTREPO_UNMOUNT_ALL:-}
```

Are you using param expansion to explicitly set this to the empty string if unset?

Reply:
Yep which is why `:-` is redundant.

Reply:
I'm having trouble finding ${param-} in the bash manual, what's it do?

Reply:
Should be under parameter expansion.
- `${foo:-X}` means 'use X if $foo is undefined or empty'
- `${foo-X}` means 'use X if $foo is undefined'

The other fancy expansions `${foo:=X}`, `${foo:?X}`, `${foo:+X}` also have colonless versions that work similarly.

Reply:
Ah, I see it now in a sentence that's hidden away. Do they expect me to actually read through the manual and not just grep through it???